File: leetcode/java/maximum-number-of-vowels-in-a-substring-of-given-length.java
import java.util.*;
class Solution{
    public int maxVowels(String s, int k){
        Set<Character> vowels = new HashSet<>();
        vowels.addAll(Arrays.asList('a', 'e', 'i', 'o','u'));
        int current_count = 0;
        for(int i = 0; i < k; i++){
            if(vowels.contains(s.charAt(i))){
                current_count+=1;
            }
        }
        int max_so_far = current_count;
        for(int i = k; i< s.length(); i++){
            if(vowels.contains(s.charAt(i))){
                current_count+=1;
            }
            if(vowels.contains(s.charAt(i-k))){
                current_count-=1;
            }
            if(current_count>max_so_far){
                max_so_far = current_count;
            }
        }
        return max_so_far;
    }
}

File: leetcode/python/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {"a", "e", "i", "o", "u"}
        l,r,count,n = 0,0,0,len(s)
        max_count = 0
        while r < k:
            if s[r] in vowels:
                count+=1
            r+=1
        max_count = max(max_count,count)
        while r < n:
            if s[l] in vowels:
                count-=1
            if s[r] in vowels:
                count+=1
            l+=1
            r+=1
            max_count = max(max_count,count)
        return max_count